frontend/src/views/housing/project/ProjectView.vue

```diff
@@ -0,0 +1,407 @@
+    permitTypes: getPermitTypes.value
+  })
+    .sort(permitNameSortFcn)
+    .sort(permitBusinessSortFcn);
+});
+const permitsSubmitted: ComputedRef<Array<CombinedPermit>> = computed(() => {
+  let firstFilter = permitFilter({
+    permitNeeded: PermitNeeded.YES,
+    permitStatus: PermitStatus.APPLIED,
+    permits: getPermits.value,
+    permitTypes: getPermitTypes.value
+  });
+  let secondFilter = permitFilter({
+    permitNeeded: PermitNeeded.YES,
+    permitStatus: PermitStatus.COMPLETED,
+    permits: getPermits.value,
+    permitTypes: getPermitTypes.value
+  });
+  return firstFilter.concat(secondFilter).sort(permitNameSortFcn).sort(permitBusinessSortFcn);
+});
+
+// Actions
+function permitBusinessSortFcn(a: CombinedPermit, b: CombinedPermit) {
+  return a.businessDomain > b.businessDomain ? 1 : -1;
+}
+function permitNameSortFcn(a: CombinedPermit, b: CombinedPermit) {
+  return a.name > b.name ? 1 : -1;
+}
+function permitFilter(config: PermitFilterConfig) {
+  const { permitNeeded, permits, permitStatus, permitTypes } = config;
+  let returnArray: Array<any> = permits;
+
+  if (permitNeeded) {
+    returnArray = returnArray.map((p) => {
+      const pType = permitTypes.find((pt) => pt.permitTypeId === p?.permitTypeId && p.needed === permitNeeded);
+      if (pType) return { ...p, ...pType };
+    });
+  }
+
+  if (permitStatus) {
+    returnArray = returnArray.map((p) => {
+      const pType = permitTypes.find((pt) => pt.permitTypeId === p?.permitTypeId && p.status === permitStatus);
+      if (pType) return { ...p, ...pType };
+    });
+  }
+
+  return returnArray.filter((pt) => !!pt) as Array<CombinedPermit>;
+}
+
+function displayTrackerStatus(authStatus: string | undefined, permitState: string | undefined): string {
+  if (permitState === PermitStatus.APPLIED) {
+    switch (authStatus) {
+      case PermitAuthorizationStatus.IN_REVIEW:
+      case PermitAuthorizationStatus.PENDING:
+        return authStatus;
+      case PermitAuthorizationStatus.NONE:
+      default:
+        return 'Submitted';
+    }
+  }
+  if (permitState === PermitStatus.COMPLETED) {
+    switch (authStatus) {
+      case PermitAuthorizationStatus.ISSUED:
+      case PermitAuthorizationStatus.DENIED:
+        return authStatus;
+      default:
+        return PermitStatus.COMPLETED;
+    }
+  }
+  return 'Submitted';
+}
+
+onMounted(async () => {
+  const [submissionValue, permitTypesValue] = (
+    await Promise.all([submissionService.getSubmission(props.submissionId), permitService.getPermitTypes()])
+  ).map((r) => r.data);
+
+  const permitsValue = (await permitService.listPermits(submissionValue.activityId)).data;
+  submissionStore.setSubmission(submissionValue);
+  submissionStore.setPermits(permitsValue);
+  typeStore.setPermitTypes(permitTypesValue);
+
+  if (submissionValue?.assignedUserId) {
+    assignee.value = (await userService.searchUsers({ userId: [submissionValue.assignedUserId] })).data[0];
+  }
+  loading.value = false;
+});
```

Comment:
Does this onMounted callback function have any form of error handling or are we expecting this to be thrown and caught up higher the vue stack? This is an async function, so if something bombs out in this body, you will get undefined behavior in the SPA application.

Reply:
Add a try catch handlers to redirect to a list page in case of errors in the submissionService, listPermits and permitService calls. Redirect on submission service call, leave on the project page for permit errors. Have a toast informing of issues for both.